### PR TITLE
Add license

### LINKED ISSUE #89: zstash needs a LICENSE!
I'm pretty sure E3SM would want there to be some sort of LICENSE file.

### PR DESCRIPTION
**Pull request by forsyth2**
Add license for `zstash`. Resolves #89.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,29 @@
+BSD 3-Clause License
+
+Copyright (c) 2018, E3SM:  Energy Exascale Earth System Model
```
Comment (forsyth2) [inline LICENSE:3]: Is the copyright year for E3SM or for `zstash`? If the latter, should it be the year `zstash` was created or the year we're adding the license?

**Review by forsyth2 — COMMENTED** [review]
Copied the license from E3SM Diags (https://github.com/E3SM-Project/e3sm_diags/blob/master/LICENSE). E3SM has a similar license (https://github.com/E3SM-Project/E3SM/blob/maint-1.0/LICENSE).

File: LICENSE (hunk 1 above)
Comment (xylar) [reply LICENSE:3]: It should be for z-stash.  It should always be the current year or a range from the year that the code was started to the current year.

Comment (xylar) [reply LICENSE:3]: I should also say that you probably can't just add a license.  You may need to consult with LLNL about the details here.

**Comment by golaz** [comment]
@forsyth2 : Dave is working on this with the lab. We should wait until that's complete.

**Comment by forsyth2** [comment]
Closing in favor of #153.

**Pull request closed without merging**